cou/steps/hypervisor.py

```diff
@@ -145,8 +157,10 @@ def azs(self) -> AZs:
             for unit in app.units.values():
                 # NOTE(rgildein): If there is no AZ, we will use empty string and all units will
                 #                 belong to a single group.
-                az = unit.machine.az or ""
-                azs[az].app_units[app.name].append(unit)
+                machine = unit.machine
+                if machine in self.machines:
```

Comment:
not-blocking: we are not using machine anywhere else, so I think we should use `if unit.machine in self.machines:`.